src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,76 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.ShoppingCartDao;
+import com.cinema.exceptions.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.User;
+import com.cinema.util.HibernateUtil;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import org.hibernate.Hibernate;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Could not insert shopping cart entity.", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaQuery<ShoppingCart> criteriaQuery = cb.createQuery(ShoppingCart.class);
+            Root<ShoppingCart> shoppingCartRoot = criteriaQuery.from(ShoppingCart.class);
+            criteriaQuery.select(shoppingCartRoot)
+                    .where(cb.equal(shoppingCartRoot.get("user"), user));
+            ShoppingCart shoppingCart = session.createQuery(criteriaQuery).uniqueResult();
+            Hibernate.initialize(shoppingCart.getTickets());
```

Comment:
you can use shoppingCartRoot.fetch() in this case. See the difference between fetch() an initialize() and make decision which one to use. I will accept any of them, but justify your choice in comment below)

Reply:
I've decided on using root.fetch(). I've found some information that by default hibernate doesn't use a second-level cache, and because of this we get the N+1 issues when using initialize() - from what I've understood, fetching is included in initialize(), so it may load the hardware more than usual fetch()

Reply:
Also, I've added SonarCloud for this project too - is it ok or should I remove it?